pkg/epp/scheduling/types/types.go

```diff
@@ -79,10 +87,15 @@ func NewContext(ctx context.Context, req *LLMRequest, pods []*PodMetrics) *Conte
 	}
 }
 
-func ToSchedulerPodMetrics(pods []backendmetrics.PodMetrics) []*PodMetrics {
-	pm := make([]*PodMetrics, 0, len(pods))
+func ToSchedulerPodMetrics(pods []backendmetrics.PodMetrics) []Pod {
+	pm := make([]Pod, 0, len(pods))
 	for _, pod := range pods {
-		pm = append(pm, &PodMetrics{pod.GetPod().Clone(), pod.GetMetrics().Clone()})
+		pm = append(pm, &PodMetrics{Pod: pod.GetPod().Clone(), Metrics: pod.GetMetrics().Clone()})
 	}
 	return pm
 }
+
+// Result captures the scheduler result.
+type Result struct {
```

Comment:
Creating a Result struct so it will be easier to extend in the future (such as adding fallback)

Reply:
I get the idea of trying to built it for future changes. I think at this point this is more confusing than helping.
for example when looking on picker interface and more specifically on the func decleration:
```
Pick(ctx *Context, pods []Pod) (*Result, error)
```
I was expecting to see that `Pod` is selected and not *Result. 
basically it's more or less the same comment as pre/post schedule.. as long as it's not used, it's not needed.

Reply:
`Result` gives us flexibility to add more, such as "reason for this pod is picked", or the "fallback pod" use case. I wanted to do this now because as we add more plugin interfaces and implementations, changing the return type will become harder. Also from a readability perspective, returning a `Result` is pretty readable IMO.

Reply:
if we decide to keep the result, I would at least rename to `SchedulingResult` or something like that, as Result is very general term.